SalesforceCommerceCloud/pwa-kit
# Client-side hot reloading / HMR

# Description



This PR adds a new feature to the runtime SDK, the one that all front-end developers love and are used to using it for granted nowadays, which is **CLIENT SIDE HOT RELOADING**!

With client side hot reloading, developers can save their changes in code editor and see the changes reflected on the browser immediately, without refreshing the web page.

See it in action: 

https://user-images.githubusercontent.com/10948652/170388489-b763ec5e-5c42-45c6-add6-d749c3acd116.mov

Related PR #379

TODO:
- [x] file 3pp request

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- installed dependencies: `@pmmmwh/react-refresh-webpack-plugin` and `react-refresh`
- add webpack plugin to the webpack configuration
- update dev server to serve modules by using webpack hot middleware

# How to Test-Drive This PR

- `npm start` in typescript template or retail react app
- see changes reflected on browser without refreshing
- refresh the browser and verify that there is no error in the console ← meaning that both client and server bundles are in sync, and that the _server-side_ hot reloading from before still works

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
